src/callbacks_step/visualization.jl

```diff
@@ -0,0 +1,223 @@
+             "variable names" => visualization_callback.variable_names,
+             "show mesh" => visualization_callback.show_mesh,
+             "plot creator" => visualization_callback.plot_creator,
+             "plot data creator" => visualization_callback.plot_data_creator,
+            ]
+    summary_box(io, "VisualizationCallback", setup)
+  end
+end
+
+
+"""
+    VisualizationCallback(; interval=0,
+                            solution_variables=cons2prim,
+                            variable_names=[],
+                            show_mesh=false,
+                            plot_data_creator=PlotData2D,
+                            plot_creator=show_plot,
+                            plot_arguments...)
+
+Create a callback that visualizes results during a simulation, also known as *in-situ
+visualization*.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+
+The `interval` specifies the number of time step iterations after which a new plot is generated. The
+available variables to plot are configured with the `solution_variables` parameter, which acts the
+same way as for the [`SaveSolutionCallback`](@ref). The variables to be actually plotted can be
+selected by providing a single string or a list of strings to `variable_names`, and if `show_mesh`
+is true, an additional plot with the mesh will be generated.
+
+To customize the generated figure, `plot_data_creator` allows to use different plot data types. With
+`plot_creator` you can further specify an own function to visualize results, which must support the
+same interface as the default implementation [`show_plot`](@ref). All remaining
+keyword arguments are collected and passed as additional arguments to the plotting command.
+"""
+function VisualizationCallback(; interval=0,
+                                 solution_variables=cons2prim,
+                                 variable_names=[],
+                                 show_mesh=false,
+                                 plot_data_creator=PlotData2D,
+                                 plot_creator=show_plot,
+                                 plot_arguments...)
+  mpi_isparallel() && error("this callback does not work in parallel yet")
+
+  if variable_names isa String
+    variable_names = String[variable_names]
+  end
+
+  visualization_callback = VisualizationCallback(interval,
+                                                 solution_variables, variable_names, show_mesh,
+                                                 plot_data_creator, plot_creator,
+                                                 Dict{Symbol,Any}(plot_arguments))
+
+  # Warn users if they create a visualization callback without having loaded the Plots package
+  if !(:Plots in names(@__MODULE__, all=true))
+    @warn "Package `Plots` not loaded but required by `VisualizationCallback` to visualize results"
+  end
```

Comment:
What about including this file via Requires only if Plots is loaded? That seems to be more clean to me.

Reply:
I thought about it as well, but decided to do it this way for now for mainly two (and a half) reasons:
1. It did not play nicely with Revise: Some changes to the callback were not picked up by Revise and it annoyed me.
2. Generally, the visualization callback might not be limited to Plots.jl but could also be used by other plotting packages.
Also, I felt it might be handy to create a visualization callback without having loaded Plots.jl yet, e.g., for testing something unrelated to plotting.

Reply:
Okay, sounds reasonable. Maybe that's worth a comment somewhere in the code?

Reply:
Done in 77e706f3.